File: zig/space-age/space_age.zig
pub const Planet = enum {
    earth,
    mercury,
    venus,
    mars,
    jupiter,
    saturn,
    uranus,
    neptune,

    pub fn age(self: Planet, seconds: usize) f64 {
        const earthSecondsPerYear: f64 = 365.25 * 24 * 60 * 60;
        const orbitalPeriod: f64 = switch (self) {
            Planet.earth => 1.0,
            Planet.mercury => 0.2408467,
            Planet.venus => 0.61519726,
            Planet.mars => 1.8808158,
            Planet.jupiter => 11.862615,
            Planet.saturn => 29.447498,
            Planet.uranus => 84.016846,
            Planet.neptune => 164.79132,
        };
        const sec: f64 = @floatFromInt(seconds);
        return sec / earthSecondsPerYear / orbitalPeriod;
    }
};

File: c/space-age/space_age.c
#include "space_age.h"

float age(planet_t planet, int64_t seconds)
{
  if ((int)planet < 0) return -1;
  float earth_seconds_per_year = 365.25 * 24 * 60 * 60;
  float orbital = 1.0;
  switch (planet) {
    case EARTH:
      break;
    case MERCURY:
      orbital = 0.2408467;
      break;
    case VENUS:
      orbital = 0.61519726;
      break;
    case MARS: 
      orbital = 1.8808158;
      break;
    case JUPITER:
      orbital = 11.862615;
      break;
    case SATURN:
      orbital = 29.447498;
      break;
    case URANUS:
      orbital = 84.016846;
      break;
    case NEPTUNE:
      orbital = 164.79132;
      break;
  }
  return (float)seconds / earth_seconds_per_year / orbital;
}
